core/src/commonMain/kotlin/com/xebia/functional/xef/embeddings/Embeddings.kt

```diff
@@ -1,8 +1,13 @@
 package com.xebia.functional.xef.embeddings
 
 import com.xebia.functional.xef.llm.models.embeddings.RequestConfig
+import kotlin.jvm.JvmStatic
 
-data class Embedding(val data: List<Float>)
+data class Embedding(val data: List<Float>) {
+  companion object {
+    @JvmStatic fun apply(data: List<Float>): Embedding = Embedding(data)
```

Comment:
I thought data classes could be instantiated the same from Java. Why do they need a static `apply`?

Reply:
See https://github.com/xebia-functional/xef/issues/275#issuecomment-1652423758

In scala, Kotlin data classes report to the compiler a final, non-default, compiler-generated `kotlin.jvm.internal.DefaultConstructorMarker` as one of their arguments. This happens even if you don't have default arugments in the data class constructor, and `@JvmOverloads` does not fix the issue.